File: backend/src/main/java/z9/hobby/domain/favorite/dto/FavoriteResponse.java
package z9.hobby.domain.favorite.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import z9.hobby.domain.favorite.entity.FavoriteEntity;

public class FavoriteResponse {
    @Getter
    @Builder
    @AllArgsConstructor
    public static class ResponseData {
        private final Long id;
        private final String favoriteName;

        public static ResponseData from(FavoriteEntity favorite) {
            return ResponseData.builder()
                    .id(favorite.getId())
                    .favoriteName(favorite.getName())
                    .build();
        }
    }
}

File: backend/src/main/java/z9/hobby/domain/favorite/dto/FavoriteResponse.kt
package z9.hobby.domain.favorite.dto;

import z9.hobby.domain.favorite.entity.FavoriteEntity

class FavoriteResponse {
    data class ResponseData(
        val id: Long?,
        val favoriteName: String
    ) {
        companion object {
            @JvmStatic
            fun from(favorite: FavoriteEntity): ResponseData {
                return ResponseData(
                    id = favorite.id,
                    favoriteName = favorite.name
                )
            }
        }
    }
}
